packages/serverpod/lib/src/server/server.dart

```diff
@@ -228,7 +228,21 @@ class Server {
       await request.response.close();
       return;
     } else if (uri.path == '/websocket') {
-      var webSocket = await WebSocketTransformer.upgrade(request);
+      WebSocket webSocket;
+      try {
+        webSocket = await WebSocketTransformer.upgrade(request);
+      } catch (e) {
+        // Handle "WebSocketException: Invalid WebSocket upgrade request".
+        stderr.writeln('Failed to upgrade connection to websocket: $e');
+        try {
+          request.response.statusCode = HttpStatus.badRequest;
+          await request.response.close();
+        } catch (e) {
+          // The statusCode setter throws StateException if the the header has
+          // already been sent. Ignore this.
+        }
```

Comment:
Suggestion: Remove the `request.response.statusCode` part as that will always fail, since the underlying library is already doing this for us! Instead only catch the exception type `WebSocketException` and ignore it (I suppose we can print the log like you do here).


Here is part of the implementation in the lib:
<img width="683" alt="Screenshot 2024-05-30 at 14 57 49" src="https://github.com/serverpod/serverpod/assets/7395515/9bed69a7-adfb-49fb-a808-c226804dba2c">

Reply:
OK, that explains why I needed the inner catch... You can remove the inner try/catch. (Sorry, I have no spare time to update PRs right now.)